app/src/main/java/ch/epfl/sweng/SDP/game/RankingFragment.java

```diff
@@ -92,35 +154,95 @@ public void onCancelled(@NonNull DatabaseError databaseError) {
 
         private final String[] players;
-
-        private RankingAdapter(Context context, String[] players) {
+        private final Integer[] rankings;
+        private final String[] trophies;
+        private final Bitmap[] drawings;
+        private final Integer[] positions;
+        
+        private RankingAdapter(Context context, String[] players, Integer[] rankings,
+                               Integer[] trophies, Bitmap[] drawings, Integer[] positions) {
             super(context, 0, players);
             this.players = players;
+            this.rankings = rankings;
+            this.trophies = RankingUtils.addSignToNumber(trophies);
+            this.drawings = drawings;
+            this.positions = positions;
         }
 
         @NonNull
         @Override
         public View getView(int position, View convertView, @NonNull ViewGroup parent) {
 
             // Check if an existing view is being reused, otherwise inflate the view
-            if (convertView == null) {
-                convertView = LayoutInflater.from(getContext())
-                        .inflate(R.layout.ranking_item, parent, false);
-            }
+            convertView = LayoutInflater.from(getContext())
+                    .inflate(R.layout.ranking_item, parent, false);
 
-            TextView pos = convertView.findViewById(R.id.position);
+            AssetManager assets = getActivity().getAssets();
+            Typeface typeMuro = Typeface.createFromAsset(assets, "fonts/Muro.otf");
+
+            // Set the font
             TextView name = convertView.findViewById(R.id.playerName);
+            name.setTypeface(typeMuro);
+            TextView ranking = convertView.findViewById(R.id.starsWon);
+            ranking.setTypeface(typeMuro);
+            TextView trophiesText = convertView.findViewById(R.id.trophiesWon);
+            trophiesText.setTypeface(typeMuro);
+
+            // Update image
+            ImageView imageview = convertView.findViewById(R.id.drawing);
+            imageview.setImageBitmap(drawings[getIndexForUserName(players[position])]);
+
+            int yellowColor = getResources().getColor(R.color.colorDrawYellow);
+            int darkColor = getResources().getColor(R.color.colorPrimaryDark);
+
+            // Set the color
+            if (!players[position].equals(account.getUsername())) {
+                name.setTextColor(yellowColor);
+                ranking.setTextColor(yellowColor);
+                trophiesText.setTextColor(yellowColor);
+                convertView.setBackgroundColor(darkColor);
+            }
 
-            pos.setText(String.format(Locale.getDefault(), "%d. ", position + 1));
-            name.setText(players[position]);
+            name.setText(String.format("%d. %s", positions[position], players[position]));
+            trophiesText.setText(this.trophies[position]);
+            ranking.setText(String.valueOf(this.rankings[position]));
```

Comment:
Extract some parts of this code to reduce it maybe

Reply:
True

Reply:
Actually I don't think it's good in this case since then I need to create multiple references the layout components